### PR TITLE
DC-1039: Remove dataset id from settings

### PR DESCRIPTION
**Pull request by rjohanek**
### Summary
Ticket: [DC-1039](https://broadworkbench.atlassian.net/browse/DC-1039)
Remove dataset_id column from snapshot_builder_settings table
Remove all entries in the snapshot_builder_settings table that don't have the snapshot_id column populated
Remove all usages of settings for datasets

[DC-1039]: https://broadworkbench.atlassian.net/browse/DC-1039?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/bio/terra/service/auth/iam/IamAction.java [hunk 1]
```diff
@@ -32,7 +32,6 @@ public enum IamAction {
   LINK_SNAPSHOT,
   UNLINK_SNAPSHOT,
   GET_SNAPSHOT_BUILDER_SETTINGS,
-  UPDATE_SNAPSHOT_BUILDER_SETTINGS,
```
Comment (snf2ye) [inline src/main/java/bio/terra/service/auth/iam/IamAction.java:35]: Thank you! (Just an extra reminder that SAM change needs to be merged first)

File: src/main/resources/api/data-repository-openapi.yaml [hunk 2]
```diff
@@ -4510,66 +4510,6 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
-  /api/repository/v1/datasets/{id}/snapshotBuilder/settings:
```
Comment (snf2ye) [inline src/main/resources/api/data-repository-openapi.yaml:4513]: Would it also be in scope for this ticket to remove the getSnapshotSnapshotBuilderSettings endpoint?

Comment (rjohanek) [reply src/main/resources/api/data-repository-openapi.yaml:4513]: Do we want to remove that endpoint? I thought we would want to keep that - at least until we add an option to the retrieveSnapshot endpoint to return a snapshot with its settings. I could file a ticket for that change if we want to do that.

File: src/main/java/bio/terra/service/auth/iam/IamAction.java (hunk 1 above)
Comment (rjohanek) [reply src/main/java/bio/terra/service/auth/iam/IamAction.java:35]: https://github.com/broadinstitute/sam/pull/1439 !

File: src/main/resources/api/data-repository-openapi.yaml (hunk 2 above)
Comment (s-rubenstein) [reply src/main/resources/api/data-repository-openapi.yaml:4513]: I think we discussed leaving `getSnapshotSnapshotBuilderSettings` as a separate endpoint/making snapshot builder settings have their own separate REST path. @pshapiro4broad might remember, I think he had thoughts here?

**Review by s-rubenstein — APPROVED** [review]
I love to see so much removed code :D

LGTM

File: src/main/resources/api/data-repository-openapi.yaml (hunk 2 above)
Comment (snf2ye) [reply src/main/resources/api/data-repository-openapi.yaml:4513]: We have a new endpoint to return the snapshot builder settings: https://data.terra.bio/swagger-ui.html#/snapshots/getSnapshotSnapshotBuilderSettings

So, I don't think we need the dataset based endpoint.

Comment (snf2ye) [reply src/main/resources/api/data-repository-openapi.yaml:4513]: I'm sorry, I was confused - I thought there was an equivalent getSnapshotBuilderSettings endpoint for datasets still hanging around. I think this PR is good as is.

**Review by snf2ye — APPROVED** [review]
LGTM! Sorry about the confusing question!

**Pull request merged**